### PR TITLE
fix MPP-3086: save location hash in cookie and restore after FXA

### PR DESCRIPTION
**Pull request by groovecoder**
This PR fixes #MPP-3086.

How to test:
1. Make an email alias
2. Sign out of Relay
3. Go to the dashboard with a link directly to the alias (e.g., https://dev.fxprivaterelay.nonprod.cloudops.mozgcp.net/accounts/profile/#1bsahsinl@mozmail.dev.fxprivaterelay.nonprod.cloudops.mozgcp.net)
4. Go thru the FXA sign-in flow
   * [ ] When you land back on the dashboard, it should focus on the mask that's in the url

- [ ] ~l10n changes have been submitted to the l10n repository, if any.~
- [ ] I've added a unit test to test for potential regressions of this bug.
- [ ] ~I've added or updated relevant docs in the docs/ directory.~
- [ ] ~All UI revisions follow the [coding standards](https://github.com/mozilla/fx-private-relay/blob/main/docs/coding-standards.md), and use Protocol tokens where applicable (see `/frontend/src/styles/tokens.scss`).~
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

**Review by maxxcrawford — APPROVED** [review]
Confirmed functionality. 

One note: Your testing steps for navigating to the dashboard from the email are incorrect:
```md
…(e.g., https://dev.fxprivaterelay.nonprod.cloudops.mozgcp.net/accounts/profile/#1bsahsinl@mozmail.dev.fxprivaterelay.nonprod.cloudops.mozgcp.net)
```

It should be URL-encoded. 
```
Go to the dashboard with a link directly to the alias (e.g., https://dev.fxprivaterelay.nonprod.cloudops.mozgcp.net/accounts/profile/#1bsahsinl%40mozmail.dev.fxprivaterelay.nonprod.cloudops.mozgcp.net)
``` 

TLDR: `@` -> `%40`

However, it looks like the server does that. This is just for testing locally.

**Review by maxxcrawford — APPROVED** [review]
Confirmed functionality. 

One note: Your testing steps for navigating to the dashboard from the email are incorrect:
```md
…(e.g., https://dev.fxprivaterelay.nonprod.cloudops.mozgcp.net/accounts/profile/#1bsahsinl@mozmail.dev.fxprivaterelay.nonprod.cloudops.mozgcp.net)
```

It should be URL-encoded. 
```
Go to the dashboard with a link directly to the alias (e.g., https://dev.fxprivaterelay.nonprod.cloudops.mozgcp.net/accounts/profile/#1bsahsinl%40mozmail.dev.fxprivaterelay.nonprod.cloudops.mozgcp.net)
``` 

TLDR: `@` -> `%40`

However, it looks like the server does that. This is just for testing locally.

**Comment by groovecoder** [comment]
Thanks @maxxcrawford also tagging @Vinnl in case there's something more about the React code I need to be mindful of ... and/or if there's a good way to test this?

File: frontend/src/pages/accounts/profile.page.tsx [hunk 1]
```diff
@@ -65,12 +66,20 @@ const Profile: NextPage = () => {
     category: "Purchase Button",
     label: "profile-bottom-promo",
   });
+  const hash = getCookie("profile-location-hash");
+  if (hash) {
+    document.location.hash = hash;
+    clearCookie("profile-location-hash");
+  }
```
Comment (Vinnl) [inline frontend/src/pages/accounts/profile.page.tsx:73]: Not a major issue since it's fairly lightweight code, but as this is a side effect, you probably only want to run it once, rather than every render:

```suggestion
  useEffect(() => {
    const hash = getCookie("profile-location-hash");
    if (hash) {
      document.location.hash = hash;
      clearCookie("profile-location-hash");
    }
  }, []);
```

**Review by Vinnl — APPROVED** [review]
Huh, I'd have sworn we did this redirect on the server side! That makes this a lot easier to implement, nice job :)

**Comment by groovecoder** [comment]
Thanks @Vinnl . I updated to `useEffect` and rebased. I'll merge when the checks pass.

**Comment by Vinnl** [comment]
@groovecoder Just a heads-up: `useEffect` takes a second argument, indicating when it should run. When you leave it out, like you did now, it will still run every render. If you pass an array, it will only run when the elements in that array change, so if you give it an empty array, it will only run once.
